File: go/leetcode/222. Count Complete Tree Nodes.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func countNodes(root *TreeNode) int {
    return count(root, 0)
}

func count(root *TreeNode, c int) int {
    if root == nil {
        return 0
    } else {
        c += 1
    }
    
    c += count(root.Right, 0)
    c += count(root.Left, 0)
    
    return c
}

File: javascript/leetcode/222. Count Complete Tree Nodes.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var countNodes = function(root) {
  return count(root, 0);
};

function count(root, c = 0) {
  if (!root) return 0;
  if (root) c++;

  c += count(root.left);
  c += count(root.right);

  return c;
}
